src/palindrom-dom.js

```diff
@@ -95,58 +94,54 @@ const PalindromDOM = (() => {
                 this.morphUrlEventHandler
             );
         }
-        
+
         /**
          * @param {String} href
-         * @throws {Error} network error if occured
-         * @fires Palindrom#palindrom-before-redirect 
-         * @fires Palindrom#palindrom-after-redirect 
-         * 
+         * @throws {Error} network error if occured or the `palindrom-before-redirect` was cancelled by calling event.preventDefault()
+         * @fires Palindrom#palindrom-before-redirect
+         * @fires Palindrom#palindrom-after-redirect
+         * @returns {Response} response (https://github.com/axios/axios#response-schema)
          */
         async getPatchUsingHTTP(href) {
             /**
              * palindrom-before-redirect event.
              *
-             * @event Palindrom#palindrom-before-redirect 
-             * @type {CustomEvent} 
+             * @event Palindrom#palindrom-before-redirect
+             * @type {CustomEvent}
              * @property {Object} detail containing `href` property that contains the URL
              */
-            const event = new CustomEvent('palindrom-before-redirect', {
+            const beforeEvent = new CustomEvent('palindrom-before-redirect', {
                 detail: {
                     href
                 },
                 cancelable: true,
                 bubbles: true
             });
-            
-            this.element.dispatchEvent(event);
-
-            // check if event was canceled
-            if(!event.defaultPrevented) {
-                let detail = {href}
-                try {
-                    await this.network.getPatchUsingHTTP(href);
-                    detail.successful = true; 
-                } catch(error) {
-                    detail.successful = false; 
-                    detail.error = error;
-                }
-                /**
-                 * palindrom-after-redirect event.
-                 *
-                 * @event Palindrom#palindrom-after-redirect
-                 * @type {CustomEvent} 
-                 * @property {Object} detail containing `{href: String, successful: boolean}` 
-                 */
-                const event = new CustomEvent('palindrom-after-redirect', {
-                    detail,
-                    bubbles: true
-                });
-                this.element.dispatchEvent(event);
-                return detail.successful;
-            } else {
-                return false;
+
+            this.element.dispatchEvent(beforeEvent);
+
+            if (beforeEvent.defaultPrevented) {
+                throw new Error(
```

Comment:
Did you consider `return Promise.reject(new Error(400));`? Then the result of the cancellation would be captured in `reject(..)`, not it `catch(..)`